quickwit/quickwit-lambda/src/indexer/ingest/helpers.rs

```diff
@@ -131,66 +137,71 @@ pub(super) async fn send_telemetry() {
     quickwit_telemetry::send_telemetry_event(TelemetryEvent::RunCommand).await;
 }
 
-pub(super) fn configure_source(
+/// Convert the incomming file path to a source config
+pub(super) async fn configure_source(
     input_path: PathBuf,
     input_format: SourceInputFormat,
     vrl_script: Option<String>,
-) -> SourceConfig {
-    let source_params = SourceParams::file(input_path);
+) -> anyhow::Result<SourceConfig> {
     let transform_config = vrl_script.map(|vrl_script| TransformConfig::new(vrl_script, None));
-    SourceConfig {
-        source_id: CLI_SOURCE_ID.to_string(),
+    let source_params = SourceParams::file(input_path.clone());
+    Ok(SourceConfig {
+        source_id: LAMBDA_SOURCE_ID.to_owned(),
         num_pipelines: NonZeroUsize::new(1).expect("1 is always non-zero."),
         enabled: true,
         source_params,
         transform_config,
         input_format,
-    }
+    })
 }
 
 /// Check if the index exists, creating or overwriting it if necessary
 pub(super) async fn init_index_if_necessary(
     metastore: &mut MetastoreServiceClient,
     storage_resolver: &StorageResolver,
-    source_config: &SourceConfig,
     default_index_root_uri: &Uri,
     overwrite: bool,
-) -> anyhow::Result<()> {
-    let checklist_result =
-        run_index_checklist(metastore, storage_resolver, &INDEX_ID, Some(source_config)).await;
-    if let Err(e) = checklist_result {
-        let is_not_found = e
-            .downcast_ref()
-            .is_some_and(|meta_error| matches!(meta_error, MetastoreError::NotFound(_)));
-        if !is_not_found {
-            bail!(e);
+) -> anyhow::Result<IndexMetadata> {
+    let metadata_result = metastore
+        .index_metadata(IndexMetadataRequest::for_index_id(INDEX_ID.clone()))
+        .await;
+    let metadata = match metadata_result {
```

Comment:
this appears to check less than `run_index_checklist` (no check on source and storage). I think skipping storage is fine because it's likely (forced?) to be the same storage as the metastore. Is the source bucket necessarily the same as the store/metastore bucket?

Reply:
You are right, we are softening the checks. `run_index_checklist` was meant to be used when indexing adhoc from the CLI. In that usecase, it's valuable to have a clear error message on the client side at the cost of an extra roundtrip to the metastore backend. When running on Lambda, this roundtrip uses up a expensive milliseconds and the check result appear in the logs of the indexing Lambda anyway, so they are not that helpful anymore.